src/frontend/src/binder/expr/function/window.rs

```diff
@@ -228,79 +230,123 @@ impl Binder {
         Ok(WindowFunction::new(kind, args, ignore_nulls, partition_by, order_by, frame)?.into())
     }
 
-    fn bind_window_frame_usize_bounds(
+    async fn bind_window_frame_usize_bounds(
         &mut self,
         start: &WindowFrameBound,
         end: Option<&WindowFrameBound>,
     ) -> Result<(FrameBound<usize>, FrameBound<usize>)> {
-        let mut convert_offset = |offset: &ast::Expr| -> Result<usize> {
```

Comment:
Can use rust's new async-closure feature to avoid having duplicated code for `start` and `end` like current code.